lib/compress/zstdmt_compress.c

```diff
@@ -528,17 +548,18 @@ static unsigned computeNbChunks(size_t srcSize, unsigned windowLog, unsigned nbT
     return (multiplier>1) ? nbChunksLarge : nbChunksSmall;
 }
 
-
-size_t ZSTDMT_compress_advanced(ZSTDMT_CCtx* mtctx,
-                               void* dst, size_t dstCapacity,
-                         const void* src, size_t srcSize,
-                         const ZSTD_CDict* cdict,
-                               ZSTD_parameters const params,
-                               unsigned overlapLog)
+static size_t ZSTDMT_compress_advanced_internal(
+                ZSTDMT_CCtx* mtctx,
+                void* dst, size_t dstCapacity,
+          const void* src, size_t srcSize,
+          const ZSTD_CDict* cdict,
+                ZSTD_CCtx_params const cctxParams,
+                unsigned overlapLog)
 {
+    ZSTD_CCtx_params const requestedParams = ZSTDMT_makeJobCCtxParams(cctxParams);
```

Comment:
OK, this is where it's a bit confusing. I believe we can clarify this.

If I do understand correctly, `ZSTDMT_makeJobCCtxParams()` creates a "souped down" version of `cctxParams`, to be used by workers. Basically it throws away anything that's not useful or misleading for workers, such as `nbThreads`, since each worker is a single thread.
So let's call it this way, this variable should be called `jobParams`, and should only used to provide parameters to worker through the `jobs` table.
For all other uses, we should continue reading values from `cctxParams`, which is our "source of truth".

Reply:
I was considering moving `ZSTDMT_makeJobCCtxParams` to `zstd_compress.c` to make it the responsibility of `zstd_compress` to pass only relevant parameters to `zstdmt`. Do you think that would make more sense?

Reply:
Not sure about this one.
I believe `zstdmt` should have all cards in hands,
the way it distributes jobs across threads, with resulting consequences on which parameters are valid or not, is its own responsibility.